### PR TITLE
[Operator] Add _upsample_bicubic2d_aa op[MooreThreads]

### PR DESCRIPTION
**Pull request by ZaccurLi**
### PR Category

Operator `_upsample_bicubic2d_aa`

### Type of Change

New Feature

### Description

add operator  `_upsample_bicubic2d_aa`

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### Performance
A100

1725542762911_32D0B79F-EE58-4034-8AC4-DD74277B9070

### CODE REVIEW

File: benchmark/test_special_perf.py [hunk 1]
```diff
@@ -140,6 +140,37 @@ def padding_kwargs(dtype, batch, size):
...
+            "input": input,
+            "output_size": output_size,
+            "align_corners": False,
+            "scales_h": None,
+            "scales_w": None,
+        }
+
+    bench = Benchmark(
+        op_name="_upsample_bicubic2d_aa",
+        # py api torch.nn.functional.interpolate,
```
Comment (Bowen12992) [inline benchmark/test_special_perf.py]: Why can't we just use the pytorch api `torch.nn.functional.interpolate` with proper `args` to make the api call the `torch._C._nn._upsample_bicubic2d_aa` functions

File: tests/test_special_ops.py [hunk 2]
```diff
@@ -418,6 +418,40 @@ def test_pad(shape, dtype, pad_mode, contiguous):
...
+        (128, 192, 42, 51),
+        (3, 7, 1023, 1025),
+    ],
+)
+@pytest.mark.parametrize("dtype", [torch.float32, torch.float16])
+def test_upsample_bicubic2d_aa(dtype, shape, scale, align_corners):
+    input = torch.rand(shape, dtype=dtype, device="cuda")
+    ref_i = to_reference(input, False)
+    output_size = tuple([int(input.shape[i + 2] * scale[i]) for i in range(2)])
+    ref_out = torch._C._nn._upsample_bicubic2d_aa(
```
Comment (Bowen12992) [inline tests/test_special_ops.py:460]: dito

File: src/flag_gems/ops/upsample_bicubic2d_aa.py [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+
+# https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L12547
+def _upsample_bicubic2d_aa(
+    input: torch.Tensor,
+    output_size: Tuple[int],
+    align_corners: bool = False,
+    scales_h: Optional[float] = None,
+    scales_w: Optional[float] = None,
+):
+    # logging.debug("GEMS UPSAMPLE BICUBIC2D AA")
```
Comment (Bowen12992) [inline src/flag_gems/ops/upsample_bicubic2d_aa.py]: uncomment this ？

File: benchmark/test_special_perf.py [hunk 4]
```diff
@@ -140,6 +140,37 @@ def padding_kwargs(dtype, batch, size):
...
+        torch_op=torch._C._nn._upsample_bicubic2d_aa,
+        arg_func=None,
+        dtypes=[torch.float16, torch.float32],
```
Comment (Bowen12992) [inline benchmark/test_special_perf.py]: Is there any problem with bfloat16

File: src/flag_gems/ops/upsample_bicubic2d_aa.py [hunk 5]
```diff
@@ -0,0 +1,525 @@
...
+                    buffer += data * weight_x[None, :]
+                weight_x_total = tl.where(weight_x_total != 0, weight_x_total, 1)
+                result += buffer / weight_x_total[None, :] * weight_y[:, None]
+            weight_y_total = tl.where(weight_y_total != 0, weight_y_total, 1)
+            result /= weight_y_total[:, None]
+            offset_o = ((n * C + c) * OH + oh[:, None]) * OW + ow[None, :]
+            tl.store(ptr_o + offset_o, result)
+
+
+# %%
```
Comment (Bowen12992) [inline src/flag_gems/ops/upsample_bicubic2d_aa.py]:
```suggestion

```

File: tests/test_special_ops.py [hunk 6]
```diff
@@ -418,6 +418,40 @@ def test_pad(shape, dtype, pad_mode, contiguous):
     gems_assert_equal(ref_out, res_out)
 
 
+@pytest.mark.parametrize("align_corners", [False, True])
```
Comment (Bowen12992) [inline tests/test_special_ops.py:443]: add `pytest.mark.{op_name}`

Comment (linkluckmars) [reply tests/test_special_ops.py:443]: done~

File: src/flag_gems/ops/upsample_bicubic2d_aa.py (hunk 5 above)
Comment (linkluckmars) [reply src/flag_gems/ops/upsample_bicubic2d_aa.py]: done

File: benchmark/test_special_perf.py (hunk 4 above)
Comment (linkluckmars) [reply benchmark/test_special_perf.py]: the bf16 type has a minor accuracy issue，like this
```
AssertionError: Tensor-likes are not close!
    Mismatched elements: 1 / 1871310 (0.0%)    
    Greatest absolute difference: 0.01171875 at index (1, 1, 265, 43) (up to 0.0025 allowed)    
    Greatest relative difference: 0.02099609375 at index (1, 1, 265, 43) (up to 0.016 allowed)  
````

File: src/flag_gems/ops/upsample_bicubic2d_aa.py (hunk 3 above)
Comment (linkluckmars) [reply src/flag_gems/ops/upsample_bicubic2d_aa.py]: fixed

File: benchmark/test_special_perf.py (hunk 1 above)
Comment (linkluckmars) [reply benchmark/test_special_perf.py]: pytorch api `torch.nn.functional.interpolate` is a superset of `torch._C._nn._upsample_bicubic2d_aa`, it's better to keep the same api with the atenlib function

File: tests/test_special_ops.py (hunk 2 above)
Comment (linkluckmars) [reply tests/test_special_ops.py:460]: ditto

**Pull request merged**